### ISSUE TITLE
全屏状态下切换竖屏异常

### ISSUE THREAD

**Issue by jzh1996** [CarGuo/GSYVideoPlayer#3751]
## 问题描述：

全屏状态下点击切换竖屏按钮，有的手机会偶发出现切换异常。点击全屏状态下的返回按钮未出现上述情况。（页面是继承StandardGSYVideoPlayer自定义，只是换了图标，没改逻辑）

## 出现问题的视频流

https://qiniu.wenluedu.com/%E9%A3%9E%E4%B9%A620230203-112046.mp4

## 问题机型/系统：

真机 荣耀畅玩30 plus

## GSY依赖版本

api "com.github.CarGuo.GSYVideoPlayer:GSYVideoPlayer:v8.3.4-release-jitpack"

## Demo中的复现步骤

DEMO->DETAIL模式->重复点击切换全屏按钮大概率复现

问题代码：(如果有)
override fun onQuitFullscreen(url: String, vararg objects: Any) {
                    super.onQuitFullscreen(url, *objects)
                    orientationUtils.backToProtVideo()
                }

## 问题log（如果有）

无

**Comment by jzh1996** [comment]
视频链接是录制的问题复现

**Comment by CarGuo** [comment]
Demo 测试了 荣耀畅玩30 plus 没复现该问题

**Comment by jzh1996** [comment]
https://video.wenluedu.com/f9c90ebb1ef0ebbaad838df1f03e9ef7.mp4      这个是用荣耀畅玩30 plus测试的DEMO复现视频，在全屏状态下用返回按钮回到竖屏时正常的，没有这个问题

**Comment by CarGuo** [comment]
测试了几次，没有复现，看你视频看起来好像是没触发旋转的代码，你可以调试下是什么情况下会触发这个问题

**Comment by jzh1996** [comment]
![image](https://user-images.githubusercontent.com/40649610/216873881-98b7588e-8c5d-4aba-8537-b2a8ea6aaa8a.png)
经过多次断点，我在图中断点位置等待0.5S左右，正常情况下即使是在断点中，手机上的页面也是回到了竖屏状态。如果等待0.5S后还是在横屏，那放开断点就会复现BUG，所以初步判断是图中选中的代码出了问题。但是我看里面的代码也就是正常设置横竖屏，没有其他操作

**Comment by jzh1996** [comment]
不是。有两种情况：第一种正常情况是在断点出等待0.5S左右，手机上的页面回到了竖屏状态，放开断点也是一切正常；第二中横竖屏切换异常情况是在断点不论等多久，手机上的页面都是在横屏状态，放开断点就出现了视频中的情况

**Comment by CarGuo** [comment]
主要是我这边暂时没找到是哪个点，会导致旋转的逻辑没生效，没出现问题····

**Comment by CarGuo** [comment]
目前我这边还是没测试出问题的点····上门你描绘的场景不知道是怎么进入到第二种情况下

**Comment by jzh1996** [comment]
我这边不管是自己项目还是在DEMO都是偶先，并且目前只在荣耀畅玩30 plus出现了这个BUG，打断点代码都是执行了的，也没报错

**Comment by renwfy** [comment]
https://user-images.githubusercontent.com/6526280/223317185-6b9537b9-7845-4ff2-8545-c2a62a750696.mp4

这个大概是什么原因

**Comment by yukiluwu** [comment]
@jzh1996  @CarGuo 这个问题我也遇到了，依赖版本是v8.3.5-release-jitpack，出问题的机型是红米K40， Android13

**Comment by wosika** [comment]
我也遇到该问题，是魅族20 pro上面出现，华为正常。
